enterprise/core-edge/src/main/java/org/neo4j/coreedge/raft/state/CoreState.java

```diff
@@ -134,57 +134,34 @@ public synchronized void notifyCommitted( long commitIndex )
-                    if ( raftLogEntry == null )
+                    RaftLogEntry entry = logEntrySupplier.get( logIndex );
+                    if( entry == null )
                     {
-                        if ( currentLogEntrySupplier == cursorLogEntrySupplier )
-                        {
-                            throw new IllegalStateException(
-                                    format( "Index %d not found in the raft log. lastToApply was: %d, append index " +
-                                            "is %d", i, lastToApply, raftLog.appendIndex() ) );
-                        }
-                        currentLogEntrySupplier = cursorLogEntrySupplier;
-                        raftLogEntry = currentLogEntrySupplier.get( i );
-                    }
-
-                    if ( raftLogEntry == null )
-                    {
-                        String failureMessage = format(
-                                "Log entry at index %d is not present in the log. " +
-                                "Please perform recovery by restarting this database instance. " +
-                                "Indexes to be applied are from %d up to %d. " +
-                                "The append index is %d, and the prev index is %d.",
-                                i, lastApplied + 1, lastToApply, raftLog.appendIndex(), raftLog.prevIndex() );
-                        log.error( failureMessage );
-                        throw new IllegalStateException( failureMessage );
+                        log.info( "Application aborted early. Log truncated?" );
+                        break;
                     }
 
-                    if ( raftLogEntry.content() instanceof DistributedOperation )
+                    if ( entry.content() instanceof DistributedOperation )
                     {
-                        DistributedOperation distributedOperation = (DistributedOperation) raftLogEntry.content();
+                        DistributedOperation distributedOperation = (DistributedOperation) entry.content();
                         progressTracker.trackReplication( distributedOperation );
                         distributedOperations.add( distributedOperation );
 
                         if ( distributedOperations.size() >= maxBatchSize )
                         {
-                            handleBatch( i, distributedOperations );
+                            handleBatch( logIndex, distributedOperations );
                         }
                     }
                     else
                     {
-                        handleBatch( i - 1 /* ignore the current entry */, distributedOperations );
-                        assertIndexIsValidAndIncrementLastApplied( i, 1 );
+                        handleBatch( logIndex - 1 /* ignore the current entry */, distributedOperations );
+                        assertIndexIsValidAndIncrementLastApplied( logIndex, 1 );
```

Comment:
@davidegrohmann Yea so what I meant was the assertIndexIsValidAndIncrementLastApplied, which only happens in the else-branch. But there seems to also be a call to that inside handleBatch. I am refactoring this anyway, so not important.